performance_manager/load_vp_events.py

```diff
@@ -0,0 +1,289 @@
+        raise exception
+
+
+def parse_args(args: List[str]) -> argparse.Namespace:
+    """parse args for running this entrypoint script"""
+    parser = argparse.ArgumentParser(description=DESCRIPTION)
+    parser.add_argument(
+        "--experimental",
+        action="store_true",
+        dest="experimental",
+        help="if set, use a sqllite engine for quicker development",
+    )
+    parser.add_argument(
+        "--seed",
+        action="store_true",
+        dest="seed",
+        help="if set, seed metadataLog table with paths",
+    )
+
+    return parser.parse_args(args)
+
+
+def get_db_engine(
+    experimental: bool, seed: bool
+) -> sqlalchemy.orm.sessionmaker:
+    """
+    create database orm objects, return engine for db operations
+    """
+    if experimental:
+        sql_engine = get_experimental_engine(echo=True)
+    else:
+        sql_engine = get_local_engine(echo=True)
+
+    SqlBase.metadata.drop_all(sql_engine)
+    SqlBase.metadata.create_all(sql_engine)
+
+    db_session = sessionmaker(bind=sql_engine)
+
+    if seed:
+        with db_session.begin() as cur:  # type: ignore
+            cur.execute(insert(MetadataLog.__table__), LOAD_PATHS)
+            cur.commit()
+
+    return db_session
+
+
+if __name__ == "__main__":
+    load_environment()
+
+    parsed_args = parse_args(sys.argv[1:])
+
+    session = get_db_engine(parsed_args.experimental, parsed_args.seed)
+
+    """
+    pull list of objects that need processing from metadata table
+    group objects by similar hourly folders
+    """
+    read_md_log = select((MetadataLog.id, MetadataLog.path)).where(
+        MetadataLog.processed == 0
+    )
+    with session.begin() as s:  # type: ignore
+        paths_to_load: dict[str, dict[str, list]] = {}
+        for path_id, path in s.execute(read_md_log):
+            path = pathlib.Path(path)
+            if path.parent not in paths_to_load:
+                paths_to_load[path.parent] = {"ids": [], "paths": []}
+            paths_to_load[path.parent]["ids"].append(path_id)
+            paths_to_load[path.parent]["paths"].append(str(path))
+
+    # for folder in list(paths_to_load.keys())[:2]:
+    for folder in paths_to_load.keys():
+        ids = paths_to_load[folder]["ids"]
+        paths = paths_to_load[folder]["paths"]
+
+        try:
+            new_events = get_vp_dataframe(paths)
+            logging.info(
+                "Size of dataframe from %s is %d", folder, new_events.shape[0]
+            )
+            new_events = transform_vp_dtyes(new_events)
+            logging.info(
+                "Size of dataframe with updated dtypes is %d",
+                new_events.shape[0],
+            )
+            new_events = transform_vp_timestamps(new_events)
+            logging.info(
+                "Size of dataframe with transformed timestamps is %d",
+                new_events.shape[0],
+            )
+
+            merge_vehicle_position_events(str(folder), new_events, session)
```

Comment:
Run ETL logic, if successful then update `metadataLog` table for processed files.